test/test_advisory.py

```diff
@@ -368,3 +369,38 @@ def test_advisory_html_urlize_description(db, client):
     assert 'qux <a href="/{0}">{0}</a> is'.format('AVG-1') in data
     assert 'bar <a href="{0}" rel="noopener">{0}</a> is'.format('https://foo.bar') in data
     assert 'lol <a href="/{0}">{0}</a>.'.format('CVE-1111-2222') in data
+
+
+@pytest.mark.parametrize('patch_get', [404], indirect=True)
+@create_package(name='foo', version='1.2.3-4')
+@create_group(id=DEFAULT_GROUP_ID, packages=['foo'], affected='1.2.3-3', fixed='1.2.3-4')
+@create_advisory(id=DEFAULT_ADVISORY_ID, group_package_id=DEFAULT_GROUP_ID, advisory_type=issue_types[1])
+@logged_in
+def test_advisory_publish_advisory_not_found(db, client, patch_get):
+    resp = client.post(url_for('tracker.publish_advisory', asa=DEFAULT_ADVISORY_ID), follow_redirects=True,
+                       data=dict(reference='https://archlinux.org', confirm=True))
+    assert 200 == resp.status_code
+    assert 'Failed to fetch advisory' in resp.data.decode()
+
+
+@pytest.mark.parametrize('patch_get', ['No advisory'], indirect=True)
+@create_package(name='foo', version='1.2.3-4')
+@create_group(id=DEFAULT_GROUP_ID, packages=['foo'], affected='1.2.3-3', fixed='1.2.3-4')
+@create_advisory(id=DEFAULT_ADVISORY_ID, group_package_id=DEFAULT_GROUP_ID, advisory_type=issue_types[1])
+@logged_in
+def test_advisory_publish_advisory_text_wrong(db, client, patch_get):
+    resp = client.post(url_for('tracker.publish_advisory', asa=DEFAULT_ADVISORY_ID), follow_redirects=True,
+                       data=dict(reference='https://archlinux.org', confirm=True))
+    assert 200 == resp.status_code
+    assert 'Failed to fetch advisory' in resp.data.decode()
+
+
+@create_package(name='foo', version='1.2.3-4')
+@create_group(id=DEFAULT_GROUP_ID, packages=['foo'], affected='1.2.3-3', fixed='1.2.3-4')
+@create_advisory(id=DEFAULT_ADVISORY_ID, group_package_id=DEFAULT_GROUP_ID, advisory_type=issue_types[1])
+@logged_in
+def test_advisory_publish_advisory(db, client, patch_get):
```

Comment:
This is the falling log, if it helps at all 

```bash
================================================= test session starts =================================================
platform linux -- Python 3.6.4, pytest-3.4.0, py-1.5.2, pluggy-0.6.0
rootdir: /home/santiago/Documents/personal/programas/arch-security-tracker, inifile:
plugins: cov-2.5.1
collected 32 items                                                                                                    

test/test_advisory.py ...............................F                                                          [100%]

====================================================== FAILURES =======================================================
___________________________________________ test_advisory_publish_advisory ____________________________________________

db = <SQLAlchemy engine=sqlite:///:memory:>, client = <FlaskClient <Flask 'tracker'>>, patch_get = None

    @create_package(name='foo', version='1.2.3-4')
    @create_group(id=DEFAULT_GROUP_ID, packages=['foo'], affected='1.2.3-3', fixed='1.2.3-4')
    @create_advisory(id=DEFAULT_ADVISORY_ID, group_package_id=DEFAULT_GROUP_ID, advisory_type=issue_types[1])
    @logged_in
    def test_advisory_publish_advisory(db, client, patch_get):
        resp = client.post(url_for('tracker.publish_advisory', asa=DEFAULT_ADVISORY_ID), follow_redirects=True,
                           data=dict(reference='https://archlinux.org', confirm=True))
        assert 200 == resp.status_code
>       assert 'Published {}'.format(DEFAULT_ADVISORY_ID) in resp.data.decode()
E       assert 'Published ASA-201802-1' in '<html>\n\t<head>\n\t\t<title>Publish ASA-201802-1 - Arch Linux</title>\n\t\t<meta http-equiv="Content-Type" content="...d="submit" name="submit" type="submit" value="publish">\n\t\t\t\t</form>\n\t\t\t</div>\n\t\t</div>\n\t</body>\n</html>'
E        +  where 'Published ASA-201802-1' = <built-in method format of str object at 0x7fbb7be1b8b0>('ASA-201802-1')
E        +    where <built-in method format of str object at 0x7fbb7be1b8b0> = 'Published {}'.format
E        +  and   '<html>\n\t<head>\n\t\t<title>Publish ASA-201802-1 - Arch Linux</title>\n\t\t<meta http-equiv="Content-Type" content="...d="submit" name="submit" type="submit" value="publish">\n\t\t\t\t</form>\n\t\t\t</div>\n\t\t</div>\n\t</body>\n</html>' = <built-in method decode of bytes object at 0x5585247b1e00>()
E        +    where <built-in method decode of bytes object at 0x5585247b1e00> = b'<html>\n\t<head>\n\t\t<title>Publish ASA-201802-1 - Arch Linux</title>\n\t\t<meta http-equiv="Content-Type" content=...d="submit" name="submit" type="submit" value="publish">\n\t\t\t\t</form>\n\t\t\t</div>\n\t\t</div>\n\t</body>\n</html>'.decode
E        +      where b'<html>\n\t<head>\n\t\t<title>Publish ASA-201802-1 - Arch Linux</title>\n\t\t<meta http-equiv="Content-Type" content=...d="submit" name="submit" type="submit" value="publish">\n\t\t\t\t</form>\n\t\t\t</div>\n\t\t</div>\n\t</body>\n</html>' = <Response 2768 bytes [200 OK]>.data

test/test_advisory.py:406: AssertionError
========================================= 1 failed, 31 passed in 3.91 seconds =========================================
```

Reply:
Can't reproduce this, hmm:

test/test_advisory.py::test_advisory_publish_advisory PASSED                                                                                            [100%]

Reply:
@SantiagoTorres can you investigate via PDB why and what is exactly the problem at your side? seems to work here and also on travis

Reply:
@SantiagoTorres ping, otherwise i simply gonna accept this